### PR TITLE
Server1

### PR DESCRIPTION
**Pull request by jrootham**
This is a major update. None of this code has been reviewed.  It is also written in Clojure.

### CODE REVIEW

File: app/server/services/test-bids.service
Comment (AcharyaS97) [inline app/server/services/test-bids.service:1]: We could change some of these setting values to build them relative to the project directory, for the sake of portability

File: app/server/toronto-bids/toronto-bids/src/toronto_bids/core.clj [hunk 1]
```diff
@@ -8,38 +8,77 @@
 	 (:require [clojure.java.jdbc :as jdbc])
 	 (:require [clojure.data.json :as json])
 	 (:require [toronto-bids.stuff :as stuff])
+	 (:require [toronto-bids.documents :as documents])
 ;	 (:require [ring-debug-logging.core :as debug])
 )
 
-(defn write-simple [table columns]
+(defn write-simple [db table columns]
```
Comment (AcharyaS97) [inline app/server/toronto-bids/toronto-bids/src/toronto_bids/core.clj:15]: Admittedly this is pedantic but could we rename this function to something like 'read-db-columns' to reflect the fact that nothing is being written to the db?

File: app/server/toronto-bids/toronto-bids/src/toronto_bids/documents.clj [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		"document.id IN (SELECT id FROM" 
+		" (SELECT id, short_description, description, search_text FROM document" 
+		" HAVING MATCH(short_description, description, search_text) AGAINST (? IN BOOLEAN MODE)) AS found)"
+	)
+)
+
+; The argument-string
+; (call_number type division commodity commodity_type posting_date_after posting_date_before closing_date_after closing_date_before buyer search-text) 
+
+(def TEST-LIST
```
Comment (AcharyaS97) [inline app/server/toronto-bids/toronto-bids/src/toronto_bids/documents.clj:19]: Rename this function to reflect that it's used to create a list of arguments document fetch query?

File: app/server/toronto-bids/toronto-bids/src/toronto_bids/documents.clj [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+(def HEAD (str "SELECT DISTINCT" COLUMN-STRING FROM-STRING))
+
+(def ORDER-STRING " ORDER BY posting_date")
+
+(def BUYER-SELECT 
+	"SELECT buyer, phone, email, location FROM document_buyer JOIN buyer ON buyer_id = buyer.id JOIN location ON location_id = location.id WHERE document_id = ?;"
+)
+
+(defn where-phrase [test start]
+	(if start (str " WHERE " test) (str " AND " test))
```
Comment (AcharyaS97) [inline app/server/toronto-bids/toronto-bids/src/toronto_bids/documents.clj:46]: To reduce branching here, could you conditionally include the 'where' keyword as part of the logic of the 'HEAD' method?

File: app/server/services/test-bids.service
Comment (jrootham) [reply app/server/services/test-bids.service:1]: The .service files are systemd config files. They run in a global setting and require absolute paths.  To make them portable they could be code generated.

File: app/server/toronto-bids/toronto-bids/src/toronto_bids/core.clj (hunk 1 above)
Comment (jrootham) [reply app/server/toronto-bids/toronto-bids/src/toronto_bids/core.clj:15]: The write is referring to the output stream.  Could change all the write names to output.  Would that be better?

File: app/server/toronto-bids/toronto-bids/src/toronto_bids/documents.clj (hunk 2 above)
Comment (jrootham) [reply app/server/toronto-bids/toronto-bids/src/toronto_bids/documents.clj:19]: Not sure what you mean here.

File: app/server/toronto-bids/toronto-bids/src/toronto_bids/documents.clj (hunk 3 above)
Comment (jrootham) [reply app/server/toronto-bids/toronto-bids/src/toronto_bids/documents.clj:46]: Don't think that would work.  If the WHERE is part of HEAD then where-phrase needs to conditionally start with blank.  Could make the HEAD include WHERE TRUE,so everything starts with AND.

Comment (jrootham) [reply app/server/toronto-bids/toronto-bids/src/toronto_bids/documents.clj:46]: I have implemented WHERE TRUE in the HEAD.

**Review by AcharyaS97 — APPROVED** [review]

**Comment by tcsullivan** [comment]
Just adding database setup instructions and a couple of tweaks to `fromxml.sql` that were necessary to re-create the DB.

**Pull request merged**